tests/test_pipx_metadata_file.py

```diff
@@ -0,0 +1,140 @@
+from pathlib import Path
+import pytest  # type: ignore
+
+from helpers import assert_not_in_virtualenv, run_pipx_cli
+import pipx.constants
+from pipx.pipx_metadata_file import PipxMetadata, PackageInfo
+from pipx.util import PipxError
+
+assert_not_in_virtualenv()
+
+
+TEST_PACKAGE1 = PackageInfo(
+    package="test_package",
+    package_or_url="test_package_url",
+    pip_args=[],
+    include_apps=True,
+    include_dependencies=False,
+    apps=["testapp"],
+    app_paths=[Path("/usr/bin")],
+    apps_of_dependencies=["dep1"],
+    app_paths_of_dependencies={"dep1": [Path("bin")]},
+    package_version="0.1.2",
+)
+TEST_PACKAGE2 = PackageInfo(
+    package="inj_package",
+    package_or_url="inj_package_url",
+    pip_args=["-e"],
+    include_apps=True,
+    include_dependencies=False,
+    apps=["injapp"],
+    app_paths=[Path("/usr/bin")],
+    apps_of_dependencies=["dep2"],
+    app_paths_of_dependencies={"dep2": [Path("bin")]},
+    package_version="6.7.8",
+)
+
+
+def test_pipx_metadata_file_create(tmp_path):
+    pipx_metadata = PipxMetadata(tmp_path)
+    pipx_metadata.main_package = TEST_PACKAGE1
+    pipx_metadata.python_version = "3.4.5"
+    pipx_metadata.venv_args = ["--system-site-packages"]
+    pipx_metadata.injected_packages = {"injected": TEST_PACKAGE2}
+
+    pipx_metadata.write()
+    del pipx_metadata
```

Comment:
Why not leave `pipx_metadata` and compare against `pipx_metadata2` directly?

Reply:
Sure.
Done.